llvm/lib/Analysis/LoopAccessAnalysis.cpp

```diff
@@ -3078,20 +3078,16 @@ const LoopAccessInfo &LoopAccessInfoManager::getInfo(Loop &L) {
   return *It->second;
 }
 void LoopAccessInfoManager::clear() {
-  SmallVector<Loop *> ToRemove;
   // Collect LoopAccessInfo entries that may keep references to IR outside the
   // analyzed loop or SCEVs that may have been modified or invalidated. At the
   // moment, that is loops requiring memory or SCEV runtime checks, as those cache
   // SCEVs, e.g. for pointer expressions.
-  for (const auto &[L, LAI] : LoopAccessInfoMap) {
+  for (const auto &[L, LAI] : make_early_inc_range(LoopAccessInfoMap)) {
```

Comment:
Not 100% sure, but I guess the reason this works out is that `DenseMap` doesn't erase any buckets when erasing an element, so the iterators stay valid? But then do we even need the early increment?

Reply:
It doesn't erase any buckets:

```cpp
  bool erase(const KeyT &Val) {
    BucketT *TheBucket = doFind(Val);
    if (!TheBucket)
      return false; // not in map.

    TheBucket->getSecond().~ValueT();
    TheBucket->getFirst() = getTombstoneKey();
    decrementNumEntries();
    incrementNumTombstones();
    return true;
  }
```

make_early_inc_range stripped, thanks!

Reply:
Sounds good, thanks